modules/gapi/include/opencv2/gapi/s11n.hpp

```diff
@@ -303,17 +305,36 @@ static GRunArg exec(cv::gapi::s11n::IIStream& is, uint32_t idx) {
 };
 
 template<typename... Types>
-cv::GCompileArgs getCompileArgs(const std::vector<char> &p) {
-    std::unique_ptr<cv::gapi::s11n::IIStream> pIs = cv::gapi::s11n::detail::getInStream(p);
+cv::util::optional<GCompileArg> tryDeserializeCompArg(const std::vector<char>& sArg) {
+    std::unique_ptr<cv::gapi::s11n::IIStream> pIs = cv::gapi::s11n::detail::getInStream(sArg);
     cv::gapi::s11n::IIStream& is = *pIs;
+
+    std::string tag;
+    is >> tag;
```

Comment:
Here you read `tag` directly from the stream, but in the serialization code, you write tag to a temporary stream and then serialize its vector.

In theory, this code should work incorrectly. Please double-check.

https://github.com/opencv/opencv/pull/18651/files#diff-7b0234dbbb6f5f5bae56c90408ad90fe57c7d3070a1ed19af9ee4c909289685fR342

Reply:
Here I have an GCompileArg as serialized vector of char, which were deserialized from the main stream and this vector of char should contain that that were deserialized in temporary stream

Reply:
Please simplify, the code shouldn't cause the confusion like mine

Reply:
Simplified, thanks!